src/resources/License.py

```diff
@@ -0,0 +1,328 @@
+import logging
+from web3 import Web3
+from web3.exceptions import LogTopicError
+
+from src.abi.PILicenseTemplate.PILicenseTemplate_client import PILicenseTemplateClient
+from src.abi.LicenseRegistry.LicenseRegistry_client import LicenseRegistryClient
+from src.abi.LicensingModule.LicensingModule_client import LicensingModuleClient
+from src.abi.IPAssetRegistry.IPAssetRegistry_client import IPAssetRegistryClient
+from src.utils.license_terms import get_license_term_by_type, PIL_TYPE
+
+# Configure logging
+logging.basicConfig(level=logging.DEBUG)  # Set to DEBUG to capture all log messages
+logger = logging.getLogger(__name__)
+
+class License:
+    def __init__(self, web3: Web3, account, chain_id):
+        self.web3 = web3
+        self.account = account
+        self.chain_id = chain_id
+
+        self.license_template_client = PILicenseTemplateClient(web3)
+        self.license_registry_client = LicenseRegistryClient(web3)
+        self.licensing_module_client  = LicensingModuleClient(web3)
+        self.ip_asset_registry_client = IPAssetRegistryClient(web3)
+
+    def _get_license_terms_id(self, license_terms):
+        logger.info(f"Getting license terms ID for: {license_terms}")
+        return self.license_template_client.getLicenseTermsId(license_terms)
+
+    def registerNonComSocialRemixingPIL(self):
+        try:
+            # Get the license terms for non-commercial social remixing PIL
+            license_terms = get_license_term_by_type(PIL_TYPE['NON_COMMERCIAL_REMIX'])
+            logger.info(f"License terms: {license_terms}")
+
+            # Check if the license terms are already registered
+            license_terms_id = self._get_license_terms_id(license_terms)
+            logger.info(f"License terms ID: {license_terms_id}")
+            if (license_terms_id is not None) and (license_terms_id != 0):
+                return {'licenseTermsId': license_terms_id}
+
+            # Build the transaction
+            transaction = self.license_template_client.build_registerLicenseTerms_transaction(
+                license_terms, {
+                    'from': self.account.address,
+                    'nonce': self.web3.eth.get_transaction_count(self.account.address),
+                    'gas': 2000000,
+                    'gasPrice': self.web3.to_wei('100', 'gwei')  # Adjusted gas price for faster processing
+                }
+            )
+            logger.info(f"Transaction: {transaction}")
```

Comment:
do we need logging?

Reply:
Technically no, but for developers who are trying to work ON (not with) the SDK, it's useful to them.